src/packagedcode/phpcomposer.py

```diff
@@ -0,0 +1,403 @@
+
+    else:
+        # use the bare repr
+        package.asserted_licenses.append(models.AssertedLicense(license=repr(licenses)))
+
+    return package
+
+
+def author_mapper(author, package):
+    """
+    Update package author and return package.
+    https://getcomposer.org/doc/04-schema.md#authors
+    The "author" is one person.
+    """
+    authors = []
+    for name, email, url in parse_person(author):
+        authors.append(models.Party(type=models.party_person, name=name, email=email, url=url))
+    package.authors = authors
+    return package
+
+
+def support_mapper(support, package):
+    """
+    Update support and bug tracking url.
+    https://getcomposer.org/doc/04-schema.md#support
+    """
+    package.support_contacts = [support.get('email')]
+    package.bug_tracking_url = support.get('issues')
+    package.code_view_url = support.get('source')
+    return package
+
+
+def repository_mapper(repos, package):
+    """
+    https://getcomposer.org/doc/04-schema.md#repositories
+    "repositories": [
+        {
+            "type": "composer",
+            "url": "http://packages.example.com"
+        },
+        {
+            "type": "composer",
+            "url": "https://packages.example.com",
+            "options": {
+                "ssl": {
+                    "verify_peer": "true"
+                }
+            }
+        },
+        {
+            "type": "vcs",
+            "url": "https://github.com/Seldaek/monolog"
+        },
+        {
+            "type": "pear",
+            "url": "https://pear2.php.net"
+        },
+        {
+            "type": "package",
+            "package": {
+                "name": "smarty/smarty",
+                "version": "3.1.7",
+                "dist": {
+                    "url": "http://www.smarty.net/files/Smarty-3.1.7.zip",
+                    "type": "zip"
+                },
+                "source": {
+                    "url": "https://smarty-php.googlecode.com/svn/",
+                    "type": "svn",
+                    "reference": "tags/Smarty_3_1_7/distribution/"
+                }
+            }
+        }
+    ]
+    """
+    if not repos:
+        return package
+    if isinstance(repos, basestring):
+        package.vcs_repository = parse_repo_url(repos)
+    elif isinstance(repos, list):
+        for repo in repos:
+            if repo.get('type') == 'vcs':
+                repo_url = repo.get('url')
+                if repo_url.startswith('svn'):
+                    package.vcs_tool = 'svn'
+                elif repo_url.startswith('hg'):
+                    package.vcs_tool = 'hg'
+                elif repo_url.startswith('cvs'):
+                    package.vcs_tool = 'cvs'
+                else:
+                    package.vcs_tool = 'git'
+                package.vcs_repository = parse_repo_url(repo.get('url'))
+    return package
+
+
+VCS_URLS = (
```

Comment:
Not listed, but the repository in composer doc mentions: 

> vcs: The version control system repository can fetch packages from git, svn, fossil and hg repositories.

https://getcomposer.org/doc/04-schema.md#repositories